build_ngtf.py

```diff
@@ -0,0 +1,400 @@
+    if (len(tf_wheel_files) != 1):
+        raise Exception(
+            "artifacts directory contains more than 1 version of tensorflow wheel"
+        )
+
+    call(["pip", "install", "-U", tf_wheel_files[0]])
+
+    import tensorflow as tf
+    cxx_abi = tf.__cxx11_abi_flag__
+    print("LIB: %s" % tf.sysconfig.get_lib())
+    print("CXX_ABI: %d" % cxx_abi)
+
+    # popd
+    os.chdir(pwd)
+
+    return str(cxx_abi)
+
+
+def build_ngraph_tf(artifacts_location, ngtf_src_loc, venv_dir, cmake_flags):
+    pwd = os.getcwd()
+
+    # Load the virtual env
+    load_venv(venv_dir)
+
+    # Get the absolute path for the artifacts
+    artifacts_location = os.path.abspath(artifacts_location)
+
+    ngtf_src_loc = os.path.abspath(ngtf_src_loc)
+    print("Source location: " + ngtf_src_loc)
+
+    os.chdir(ngtf_src_loc)
+
+    # mkdir build directory
+    path = 'build'
+    try:
+        os.makedirs(path)
+    except OSError as exc:  # Python >2.5
+        if exc.errno == errno.EEXIST and os.path.isdir(path):
+            pass
+
+    # Run cmake
+    os.chdir('build')
+    cmake_cmd = ["cmake"]
+    cmake_cmd.extend(cmake_flags)
+    cmake_cmd.extend([ngtf_src_loc])
+    print("NGRAPH_TF Cmake options: %s" + str(cmake_cmd))
+    if (call(cmake_cmd) != 0):
+        raise Exception("Error running cmake command: " + str(cmake_cmd))
+
+    if (call(["make", "-j", "install"]) != 0 ):
+        raise Exception("Error running make command ")
+
+    os.chdir(os.path.join("python", "dist"))
+    ngtf_wheel_files = glob.glob("ngraph_config-*.whl")
+    if (len(ngtf_wheel_files) != 1):
+        raise Exception(
+            "Error getting the ngraph-tf wheel file"
+        )
+    
+    output_wheel = ngtf_wheel_files[0]
+    print( "OUTPUT WHL FILE: %s" % output_wheel)
+    
+    output_path = os.path.join(artifacts_location, output_wheel)
+    print( "OUTPUT WHL DST: %s" % output_path)
+    # Delete just in case it exists
+    try:
+        os.remove(output_path)
+    except OSError:
+        pass
+
+    # Now copy
+    shutil.copy2(output_wheel, artifacts_location)
+
+    os.chdir(pwd)
+    return output_wheel
+
+def install_ngraph_tf(venv_dir, ngtf_pip_whl):
+    # Load the virtual env
+    load_venv(venv_dir)
+
+    # Intall the ngtf_wheel
+    call(["pip", "install", "-U", ngtf_pip_whl])
+
+    import tensorflow as tf; 
+    print('TensorFlow version: r',tf.__version__); 
+    print(tf.__compiler_version__);
+    import ngraph_config; print(ngraph_config.__version__)
+
+    pass
+
+def download_repo(target_name, repo, version):
+
+    # First download to a temp folder
+    call(["git", "clone", repo, target_name])
+
+    # Next goto this folder nd determone the name of the root folder
+    pwd = os.getcwd()
+
+    # Go to the tree
+    os.chdir(target_name)
+
+    # checkout the specified branch
+    call(["git", "checkout", version])
+
+    os.chdir(pwd)
+
+def main():
+    parser = argparse.ArgumentParser()
```

Comment:
We could add a helpline:
`description='Builds TF, ngraph, and ngraph-tf (not TF embedded ngraph) for python 3'`